src/Builder/FrontendDialectTransformer.cpp

```diff
@@ -947,7 +938,7 @@ class FrontendGenImpl {
     auto *fnEntryBlock = funcOp.addEntryBlock();
 
     // Save caller context, while generating callee function body.
-    SymbolMapping callerScope(std::move(frontend_symbols_));
+    frontend_symbols_.pushScope(func_name_prefix);
```

Comment:
Just for my understanding: what exactly does this scope mechanism do? Strictly speaking, the callee function body is not a nested-scope and it cannot reference symbols defined in the caller function. Will this potentially cause an issue (in how erroneous input models are handled ... I guess it may not matter in a correct model)?

Reply:
Sorry I was mistaken. I assumed the scoping rules of the function body are similar to those of the subgraphs associated with control-flow operations like Loop and conditionals.

It seems to me that here we need a fresh scope with only function input variables in the symbol table.

Reply:
Do you mean switch back to the previous implementation? Anyway this seems orthogonal to the main goal of the PR. Or, do you mean some other approach?

Reply:
I will switch back to your original implementation shortly. I don't see a need for another approach presently, the ad-hoc symbol table works reasonably well under most circumstances.